detections/endpoint/windows_file_and_directory_enable_readonly_permissions.yml

```diff
@@ -0,0 +1,74 @@
+name: Windows File and Directory Enable ReadOnly Permissions
+id: 1ae407b0-a042-4eb0-834a-590da055575e
+version: 1
+date: '2024-12-13'
+author: Teoderick Contreras, Splunk
+data_sources:
+- Sysmon EventID 1
+- Windows Event Log Security 4688
+type: TTP
+status: production
+description: The following analytic detects instances where file or folder permissions are modified to grant read-only access. Such changes are characterized by the presence of read-related permissions (e.g., R, REA, RA, RD) and the absence of write (W) or execute (E) permissions. Monitoring these events is crucial for tracking access control changes that could be intentional for restricting access or indicative of malicious behavior. Alerts generated by this detection help ensure that legitimate security measures are enforced while unauthorized changes are promptly investigated.
+kind: endpoint
+search: '| tstats `security_content_summariesonly` min(_time) as firstTime max(_time) as lastTime from datamodel=Endpoint.Processes 
+  where Processes.process_name IN( "icacls.exe", "cacls.exe", "xcacls.exe") AND Processes.process IN ("*/grant*", "*/G*") AND Processes.process IN ("*SYSTEM*", "*admin*", "*S-1-1-0*", "*EVERYONE*")
```

Comment:
Is there a specific reason we only focus on these groups for read only? I think any file being set to read only is interesting.